### ISSUE TITLE
device-method CLI failed because upstream does not have RestApiClient defined

### ISSUE THREAD

**Issue by compulim** [Azure/iothub-explorer#7]
In [`azure-iothub/lib/client.js`](https://github.com/Azure/azure-iot-sdk-node/blob/master/service/lib/client.js), if the service client is instantiated using `fromSharedAccessSignature`, the client does not contains `this._restApiClient`.

Thus, `device-method` CLI will fail with the following error.

Looks like it could be problem with the upstream library.

```
C:\Users\JohnDoe\Source\Repos\codeaholics-iot-demo\cli>iothub-explorer device-method hallwayLight turnOn
C:\Users\JohnDoe\AppData\Roaming\npm\node_modules\iothub-explorer\node_modules\azure-iothub\lib\device_method.js:73
  this._client.executeApiCall('POST', path, headers, this.params, this.params.timeoutInSeconds * 1000, done);
              ^

TypeError: Cannot read property 'executeApiCall' of undefined
    at DeviceMethod.invokeOn (C:\Users\JohnDoe\AppData\Roaming\npm\node_modules\iothub-explorer\node_modules\azure-iothub\lib\device_method.js:73:15)
    at Client.invokeDeviceMethod (C:\Users\JohnDoe\AppData\Roaming\npm\node_modules\iothub-explorer\node_modules\azure-iothub\lib\client.js:240:10)
    at Object.<anonymous> (C:\Users\JohnDoe\AppData\Roaming\npm\node_modules\iothub-explorer\iothub-explorer-device-method.js:41:8)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.runMain (module.js:605:10)
    at run (bootstrap_node.js:420:7)
    at startup (bootstrap_node.js:139:9)
    at bootstrap_node.js:535:3
```

**Comment by pierreca** [comment]
Thanks for the feedback @compulim, that's a bug. I'll have a fix in the next few hours and I'll update the issue when a new package is published.

**Comment by pierreca** [comment]
Fixed it here: https://github.com/Azure/azure-iot-sdk-node/commit/672bd8cae5969ed1558a7beeb1073aefeade863c

We'll release a new set of packages, I'll link them here and close the issue when done.

**Comment by pierreca** [comment]
The new package with the fix has been released (1.1.4)
https://www.npmjs.com/package/iothub-explorer
